pkg/storaged/block/create-pages.jsx

```diff
@@ -0,0 +1,107 @@
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import client from "../client";
+
+import { get_fstab_config } from "../filesystem/utils.jsx";
+
+import { make_partition_table_page } from "../partitions/partition-table.jsx";
+import { make_legacy_vdo_page } from "../legacy-vdo/legacy-vdo.jsx";
+
+import { make_unrecognized_data_card } from "./unrecognized-data.jsx";
+import { make_unformatted_data_card } from "./unformatted-data.jsx";
+import { make_locked_encrypted_data_card } from "../crypto/locked-encrypted-data.jsx";
+import { make_filesystem_card } from "../filesystem/filesystem.jsx";
+import { make_lvm2_physical_volume_card } from "../lvm2/physical-volume.jsx";
+import { make_mdraid_disk_card } from "../mdraid/mdraid-disk.jsx";
+import { make_stratis_blockdev_card } from "../stratis/blockdev.jsx";
+import { make_swap_card } from "../swap/swap.jsx";
+import { make_encryption_card } from "../crypto/encryption.jsx";
+
+import { new_page } from "../pages.jsx";
+
+/* CARD must have page_name, page_location, and page_size set.
+ */
+
+export function make_block_page(parent, block, card) {
+    let is_crypto = block.IdUsage == 'crypto';
+    let content_block = is_crypto ? client.blocks_cleartext[block.path] : block;
+    const fstab_config = get_fstab_config(content_block || block, true);
+
+    const block_stratis_blockdev = client.blocks_stratis_blockdev[block.path];
+    const block_stratis_stopped_pool = client.blocks_stratis_stopped_pool[block.path];
+    const legacy_vdo = client.legacy_vdo_overlay.find_by_backing_block(block);
+
+    const is_stratis = ((content_block && content_block.IdUsage == "raid" && content_block.IdType == "stratis") ||
+                        (block_stratis_blockdev && client.stratis_pools[block_stratis_blockdev.Pool]) ||
+                        block_stratis_stopped_pool);
+
+    if (client.blocks_ptable[block.path]) {
+        make_partition_table_page(parent, block, card);
+        return;
+    }
+
+    if (legacy_vdo) {
+        make_legacy_vdo_page(parent, legacy_vdo, block, card);
+        return;
+    }
+
+    // Adjust for encryption leaking out of Stratis
+    if (is_crypto && is_stratis) {
+        is_crypto = false;
+        content_block = block;
+    }
+
+    if (is_crypto)
+        card = make_encryption_card(card, block);
+
+    if (!content_block) {
+        // assert(is_crypto);
+        if (fstab_config.length > 0) {
+            card = make_filesystem_card(card, block, null, fstab_config);
+        } else {
+            card = make_locked_encrypted_data_card(card, block);
+        }
```

Comment:
`content_block` can only be null when `is_crypto` is true. This follows from `block` never being null, and `content_block == block` unless `is_crypto` is true.

So this can only go wrong when we change the code, but that of course is a real possibility, so I'll put in the console.error.